ports/atmel-samd/common-hal/pwmio/PWMOut.c

```diff
@@ -97,32 +79,14 @@ void pwmout_reset(void) {
     }
     Tcc *tccs[TCC_INST_NUM] = TCC_INSTS;
     for (int i = 0; i < TCC_INST_NUM; i++) {
-        if (never_reset_tc_or_tcc[TC_INST_NUM + i] > 0) {
+        if (!timer_ok_to_reset(i, false)) {
             continue;
         }
-        // Disable the module before resetting it.
-        if (tccs[i]->CTRLA.bit.ENABLE == 1) {
-            tccs[i]->CTRLA.bit.ENABLE = 0;
-            while (tccs[i]->SYNCBUSY.bit.ENABLE == 1) {
-            }
-        }
         uint8_t mask = 0xff;
         for (uint8_t j = 0; j < tcc_cc_num[i]; j++) {
             mask <<= 1;
         }
         tcc_channels[i] = mask;
```

Comment:
can the loop be eliminated, giving `tcc_channels[i].mask = 0xff << tcc_cc_num[i];` ?  (This code does have to remain here, as tcc_channels[] is only used in PWMOut)

(I don't know what the significance of these values is)